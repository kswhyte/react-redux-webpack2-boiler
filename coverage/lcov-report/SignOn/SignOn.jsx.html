<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for SignOn/SignOn.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../prettify.css" />
    <link rel="stylesheet" href="../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../index.html">All files</a> / <a href="index.html">SignOn</a> SignOn.jsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">34.62% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>9/26</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">33.33% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>5/15</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>3/6</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">34.62% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>9/26</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78
79
80
81
82
83
84
85
86
87
88
89
90
91
92
93
94
95
96
97
98
99
100
101
102
103
104
105
106
107
108
109
110
111
112
113
114
115
116
117
118
119
120
121
122
123
124
125
126
127
128
129
130
131
132
133
134
135
136
137
138
139</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">8x</span>
<span class="cline-any cline-yes">8x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">8x</span>
<span class="cline-any cline-yes">8x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">8x</span>
<span class="cline-any cline-yes">8x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">8x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { PropTypes } from 'prop-types';
import isEmail from '../../helpers/isEmail';
&nbsp;
import './sign-on.css';
&nbsp;
const propTypes = {
  submitLogin: PropTypes.func.isRequired,
  validationError: PropTypes.func,
  validationMessage: PropTypes.string
};
&nbsp;
class SignOn extends Component {
  constructor(props) {
    super(props);
    this.state = {
      loginEmail: '',
      rememberMeValue: 'off'
    };
    this.rememberMe = this.rememberMe.bind(this);
    this.handleInputEmailChange = this.handleInputEmailChange.bind(this);
  }
&nbsp;
  componentDidMount() {
    const storedUserEmail = localStorage.getItem('userEmail');
    <span class="missing-if-branch" title="if path not taken" >I</span>if (storedUserEmail !== null &amp;&amp; <span class="branch-1 cbranch-no" title="branch not covered" >storedUserEmail !== undefined </span>&amp;&amp; <span class="branch-2 cbranch-no" title="branch not covered" >storedUserEmail !== '')</span> {
<span class="cstat-no" title="statement not covered" >      this.setState({</span>
        loginEmail: storedUserEmail,
        rememberMeValue: 'on'
      });
    }
  }
&nbsp;
<span class="fstat-no" title="function not covered" >  ha</span>ndleInputEmailChange(e){
<span class="cstat-no" title="statement not covered" >     this.setState({ loginEmail: e.target.value });</span>
  }
&nbsp;
<span class="fstat-no" title="function not covered" >  re</span>memberMe() {
    let userEmailValue = <span class="cstat-no" title="statement not covered" >escape(document.querySelector('#loginEmail').value);</span>
<span class="cstat-no" title="statement not covered" >    if (this.state.rememberMeValue === 'off') {</span>
<span class="cstat-no" title="statement not covered" >      this.setState({ rememberMeValue: 'on' });</span>
<span class="cstat-no" title="statement not covered" >      localStorage.setItem('userEmail', userEmailValue);</span>
    } else {
<span class="cstat-no" title="statement not covered" >      this.setState({ rememberMeValue: 'off' });</span>
<span class="cstat-no" title="statement not covered" >      localStorage.removeItem('userEmail');</span>
    }
  }
&nbsp;
  render() {
    return (
      &lt;div className="sign-on-wrapper"&gt;
        &lt;img src="images/pattern-dots.png" className="dots-pattern" /&gt;
        &lt;div className="row"&gt;
          &lt;h2&gt;SIGN ON&lt;/h2&gt;
          &lt;div className="page-instructions col-xs-offset-4 col-xs-4"&gt;
            &lt;p&gt;Please enter your email address and password below.&lt;/p&gt;
          &lt;/div&gt;
        &lt;/div&gt;
        {this.props.validationMessage &amp;&amp;
<span class="branch-1 cbranch-no" title="branch not covered" >          &lt;div className="sm-spacer validation-error"&gt;</span>
            &lt;h5&gt;
              {this.props.validationMessage}
            &lt;/h5&gt;
          &lt;/div&gt;}
        &lt;form
          onSubmit={<span class="fstat-no" title="function not covered" >e </span>=&gt; {
<span class="cstat-no" title="statement not covered" >            e.preventDefault();</span>
            const loginEmail = <span class="cstat-no" title="statement not covered" >escape(document.querySelector('#loginEmail').value);</span>
            const loginPassword = <span class="cstat-no" title="statement not covered" >escape(document.querySelector('#loginPassword').value);</span>
            const emailValidation = <span class="cstat-no" title="statement not covered" >isEmail(loginEmail);</span>
<span class="cstat-no" title="statement not covered" >            if (!emailValidation) {</span>
<span class="cstat-no" title="statement not covered" >              this.props.validationError(</span>
                'The information entered is not recognized. Please enter a valid username and password.'
              );
<span class="cstat-no" title="statement not covered" >              return;</span>
            }
            const data = <span class="cstat-no" title="statement not covered" >{</span>
              loginEmail,
              loginPassword
            };
<span class="cstat-no" title="statement not covered" >            this.props.submitLogin(data);</span>
          }}
        &gt;
          &lt;div className="row md-spacer"&gt;
            &lt;div className="col-sm-6 input-row"&gt;
              &lt;label forHtml="loginEmail"&gt;USER NAME (EMAIL)&lt;/label&gt;
              &lt;input
                className={
                  this.props.validationMessage ? <span class="branch-0 cbranch-no" title="branch not covered" >'input-error form-group hg-input v2' </span>: 'form-group hg-input v2'
                }
                value={this.state.loginEmail}
                onChange={this.handleInputEmailChange}
                type="text"
                id="loginEmail"
                placeholder="Your Email Address"
                required
              /&gt;
            &lt;/div&gt;
            &lt;div className="col-sm-6 input-row"&gt;
              &lt;label forHtml="loginPassword"&gt;PASSWORD&lt;/label&gt;
              &lt;input
                id="loginPassword"
                type="password"
                className="form-group hg-input v2"
                placeholder="Your Password"
                label="Your Password"
                required
              /&gt;
            &lt;/div&gt;
          &lt;/div&gt;
          &lt;div className="row lg-spacer"&gt;
            &lt;div className="col-xs-offset-3 col-xs-6"&gt;
              &lt;button className="primary"&gt;SIGN ON&lt;/button&gt;
              &lt;div className="row remember-me-row"&gt;
                &lt;input
                  className="remember-checkbox"
                  checked={this.state.rememberMeValue === 'on' ? <span class="branch-0 cbranch-no" title="branch not covered" >'checked' </span>: ''}
                  onClick={this.rememberMe}
                  id="remember-checkbox"
                  type="checkbox"
                /&gt;
                &lt;label htmlFor="remember-checkbox"&gt;Remember Me&lt;/label&gt;
              &lt;/div&gt;
              &lt;div className="row forgot-password-row"&gt;
                &lt;Link to="/resetpassword"&gt;Forgot your password?&lt;/Link&gt;
              &lt;/div&gt;
            &lt;/div&gt;
          &lt;/div&gt;
        &lt;/form&gt;
      &lt;/div&gt;
    );
  }
}
&nbsp;
SignOn.propTypes = propTypes;
&nbsp;
export default SignOn;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Wed Aug 09 2017 12:07:41 GMT-0600 (MDT)
</div>
</div>
<script src="../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../sorter.js"></script>
</body>
</html>
