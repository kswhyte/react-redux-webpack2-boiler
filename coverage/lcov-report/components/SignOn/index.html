<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for components/SignOn</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> components/SignOn
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">42.42% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>14/33</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">43.48% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>10/23</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">57.14% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>4/7</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">42.42% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>14/33</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file low" data-value="SignOn.jsx"><a href="SignOn.jsx.html">SignOn.jsx</a></td>
	<td data-value="42.42" class="pic low"><div class="chart"><div class="cover-fill" style="width: 42%;"></div><div class="cover-empty" style="width:58%;"></div></div></td>
	<td data-value="42.42" class="pct low">42.42%</td>
	<td data-value="33" class="abs low">14/33</td>
	<td data-value="43.48" class="pct low">43.48%</td>
	<td data-value="23" class="abs low">10/23</td>
	<td data-value="57.14" class="pct medium">57.14%</td>
	<td data-value="7" class="abs medium">4/7</td>
	<td data-value="42.42" class="pct low">42.42%</td>
	<td data-value="33" class="abs low">14/33</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Wed Aug 09 2017 13:06:30 GMT-0600 (MDT)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
