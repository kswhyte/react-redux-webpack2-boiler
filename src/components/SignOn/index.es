import Component from './SignOn';
export default Component;
