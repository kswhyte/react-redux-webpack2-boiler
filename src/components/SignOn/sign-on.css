@import "imports/_helpers.css";
.sign-on-wrapper {
    padding-top: 40px;
    text-align: center;
    & .input-row {
        padding: 0 2rem;
    }
    & h2 {
        font-size: 4rem;
    }
    & .remember-checkbox {
        border: solid 1px black;
        width: 1rem;
        height: 1rem;
        display: inline-block;
        margin-right: 10px;
    }

    & label {
        color: var(--light-grey);
    }

    & .remember-checkbox > .checkaroo {
        margin-top: -.5rem;
        margin-left: -.2rem;
    }

    & .checkaroo > i {
        color: var(--cool-blue);
        font-size: 1.7rem;
        -webkit-text-stroke: 3px white;
    }
}
