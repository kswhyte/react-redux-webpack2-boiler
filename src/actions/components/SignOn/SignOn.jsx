import React from 'react';

import './sign-on.css';

const SignOn = () => {
  return (
    <div className="sign-on-wrapper">
      <h2>SIGN ON</h2>
    </div>
  );
};

SignOn.propTypes = {};

export default SignOn;
