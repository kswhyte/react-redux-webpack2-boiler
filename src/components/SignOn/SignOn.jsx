import React from 'react';
import { Link } from 'react-router-dom';
import { PropTypes } from 'prop-types';

import TextInput from '@hg/three-ui/HgInputsV2/TextInput';
import HgButton from '@hg/three-ui/HgButton';

import isEmail from '../../helpers/isEmail';

import './sign-on.css';

const propTypes = {
  submitLogin: PropTypes.func.isRequired,
  validationError: PropTypes.func,
  validationMessage: PropTypes.string
};

const SignOn = (props) => {
  return (
    <div className="sign-on-wrapper">
        <div className="row">
            <h2>SIGN ON</h2>
            <div className="page-instructions col-xs-offset-4 col-xs-4">
                <h4>Please enter your email address and password below.</h4>
            </div>
        </div>
        {
            (props.validationMessage) && (
                <div className="sm-spacer validation-error">
                    <h5>{props.validationMessage}</h5>
                </div>
            )
        }
        <form
            onSubmit={(e) => {
                e.preventDefault();
                const loginEmail = document.getElementById("loginEmail").value;
                const loginPassword = document.getElementById("loginPassword").value;
                const emailValidation = isEmail(loginEmail);
                if (!emailValidation) {
                  props.validationError("Must enter a valid email.");
                  return;
                }
                const data = {
                    loginEmail,
                    loginPassword
                }
                props.submitLogin(data);
            }}
          >
            <div className="row md-spacer">
                <div className="col-sm-6 input-row">
                   <TextInput
                        id="loginEmail"
                        className="form-group"
                        placeholder={'Your Email Address'}
                        label="User Name (Email)"
                        required={true}
                    />
                </div>
                <div className="col-sm-6 input-row">
                    <input
                        id="loginPassword"
                        type="password"
                        className="form-group hg-input v2"
                        placeholder="Your Password"
                        label="Your Password"
                        required
                    />
                </div>
            </div>
            <div className="row lg-spacer">
                <div className="col-xs-offset-3 col-xs-6 flex-column">
                    <HgButton
                        className="primary"
                        text="SIGN ON"
                    />
                    <div className="row sm-spacer">
                        <div className="remember-checkbox">
                            <div className="checkaroo">
                                <i className="fa fa-check" aria-hidden="true"></i>
                            </div>
                        </div>
                        <label>Remember Me</label>
                    </div>
                    <div className="row md-spacer forgot-password">
                        <Link to="/login/resetpassword">Forgot your password?</Link>
                    </div>
                </div>
            </div>
        </form>
    </div>
  );
};

SignOn.propTypes = propTypes;

export default SignOn;
