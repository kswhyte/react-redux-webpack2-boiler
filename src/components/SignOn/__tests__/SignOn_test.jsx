import React from 'react';
// import { storiesOf, action, linkTo } from '@kadira/storybook';
import SignOn from '../SignOn';
import sessionActions from '../../../actions/SessionActions';

// import { specs, describe, it } from 'storybook-addon-specifications';
import { storiesOf, describe, it, specs } from '../../../../.storybook/facade';

import { shallow, mount } from 'enzyme';
import expect, { createSpy, spyOn, isSpy } from 'expect';
import { StaticRouter } from 'react-router';

const stories = storiesOf('SignOn', module);

stories.add('Sign On Page Elements', () => {
  const props = {
    validationMessage: 'Please enter a valid email'
  };

  const signOnStory = (
    <StaticRouter {...props} context={{}}>
      <SignOn />
    </StaticRouter>
  );

  specs(() =>
    describe('SignOn Page elements and content', () => {
      it('Should render the SignOn component without crashing', () => {
        let output = mount(signOnStory);
        expect(output.find('.sign-on-wrapper').length).toEqual(1);
      });

      it('Should have a header element that says "Sign On"', () => {
        let output = mount(signOnStory);
        expect(output.find('h2').text()).toContain('SIGN ON');
      });
      it('Should have a <p> tag that says "please enter your email address and password below"', () => {
        let output = mount(signOnStory);
        expect(output.find('p').text()).toContain('Please enter your email address and password below');
      });
      it('Should have two input form fields', () => {
        let output = mount(signOnStory);
        expect(output.find('input').length).toEqual(2);
      });
      it('Should have two input form fields: username/email and password', () => {
        let output = mount(signOnStory);
        expect(output.find('#loginEmail').length).toEqual(1);
        expect(output.find('#loginPassword').length).toEqual(1);
      });
      it('Should have one call-to-action submit button', () => {
        let output = mount(signOnStory);
        expect(output.find('button').length).toEqual(1);
        expect(output.find('.hg-button').length).toEqual(1);
      });
      it('Should have one link: "Forgot Password"', () => {
        let output = mount(signOnStory);
        expect(output.find('a').length).toEqual(1);
        expect(output.find('a').text()).toContain('Forgot your password?');
      });
    })
  );
  return signOnStory;
});

stories.add('Clicking Submit', () => {
  const props2 = {
    validationMessage: 'Please enter a valid email'
  };

  const signOnStory2 = (
    <StaticRouter context={{}}>
      <SignOn {...props2} />
    </StaticRouter>
  );

  specs(() =>
    describe('Clicking Submit', () => {
      it('Should display an error message if email info is incorrect', () => {
        let output = mount(signOnStory2);
        expect(output.find('h5').text()).toContain('Please enter a valid email');
        // TODO: need to add functionality for recovering password after 2nd attempt; use a counter
      });
      it('Should get an error message and message asking if you would like to recover your password after your second attempt', () => {
        let output = mount(signOnStory2);
        // TODO: need to add functionality for recovering password after 2nd attempt; use a counter
      });
      it('Should send post with an object that contains username, password to endpoint', () => {
        let output = mount(signOnStory2);
        const userData = {
          loginEmail: 'newagent@mailinator.com',
          loginPassword: 'newpassword'
        };
        let currentRoute = window.location;

        sessionActions.startLoginClick(userData);
        // TODO: think about how to best check that the data is sent through with "startLoginClick(data)"
      });
      it('Should display red text and red bottom border for username and password if the form was submitted without any text added to each input field', () => {
        let output = mount(signOnStory2);
        // TODO: need to add error message at top of screen to indicate user has not filled out required fields
      });
    })
  );
  return signOnStory2;
});

stories.add('Remember Me Checkbox', () => {
  const props3 = {
    validationMessage: 'Please enter a valid email'
  };

  const signOnStory3 = (
    <StaticRouter context={{}}>
      <SignOn {...props3} />
    </StaticRouter>
  );

  let createSpy = expect.createSpy;
  let spyOn = expect.spyOn;
  let isSpy = expect.isSpy;

  specs(() =>
    describe('Clicking "Remember Me" checkbox', () => {
      it('Should store username and NOT the password in local storage, if not found in local storage', () => {
        let output = mount(signOnStory3);
        let loginInfo = {
          loginEmail: 'newagent@mailinator.com',
          loginPassword: 'newpassword'
        };

        let spy = expect.createSpy();
        window.localStorage.setItem('loginInfo', loginInfo);

        spy();
        output.find('.fa-check').simulate('click');
        output.find('.hg-button').simulate('click');
        expect(spy).toHaveBeenCalled();
        expect(spy.calls.length).toEqual(1);
        // expect(window.localStorage.getItem('')).toEqual('');

        spy.restore();
        expect.restoreSpies();
        // TODO: need to actually set localStorage, then check it for added loginInfo credentials
      });
      it('Should check local storage; If the user login username is there, the username input field should be filled out', () => {
        let output = mount(signOnStory3);
        expect(output.find('#loginPassword').text()).toNotExist();
        expect(output.find('#loginEmail').text()).toNotExist();

        // TODO: check that these input fields don't have text, then that they do
        // expect(output.find('#loginPassword').text()).toExist();
        // expect(output.find('#loginEmail').text()).toExist();
      });
    })
  );

  return signOnStory3;
});
