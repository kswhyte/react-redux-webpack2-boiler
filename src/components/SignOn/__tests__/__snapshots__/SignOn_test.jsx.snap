// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Clicking Submit 1`] = `
<div
  className="sign-on-wrapper"
>
  <div
    className="row"
  >
    <h2>
      SIGN ON
    </h2>
    <div
      className="page-instructions col-xs-offset-4 col-xs-4"
    >
      <p>
        Please enter your email address and password below.
      </p>
    </div>
  </div>
  <div
    className="sm-spacer validation-error"
  >
    <h5>
      Please enter a valid email
    </h5>
  </div>
  <form
    onSubmit={[Function]}
  >
    <div
      className="row md-spacer"
    >
      <div
        className="col-sm-6 input-row"
      >
        <div />
      </div>
      <div
        className="col-sm-6 input-row"
      >
        <input
          className="form-group hg-input v2"
          id="loginPassword"
          label="Your Password"
          placeholder="Your Password"
          required={true}
          type="password"
        />
      </div>
    </div>
    <div
      className="row lg-spacer"
    >
      <div
        className="col-xs-offset-3 col-xs-6 flex-column"
      >
        <div />
        <div
          className="row sm-spacer"
        >
          <div
            className="remember-checkbox"
          >
            <div
              className="checkaroo"
            >
              <i
                aria-hidden="true"
                className="fa fa-check"
              />
            </div>
          </div>
          <label>
            Remember Me
          </label>
        </div>
        <div
          className="row md-spacer forgot-password"
        >
          <a
            href="/resetpassword"
            onClick={[Function]}
          >
            Forgot your password?
          </a>
        </div>
      </div>
    </div>
  </form>
</div>
`;

exports[`Remember Me Checkbox 1`] = `
<div
  className="sign-on-wrapper"
>
  <div
    className="row"
  >
    <h2>
      SIGN ON
    </h2>
    <div
      className="page-instructions col-xs-offset-4 col-xs-4"
    >
      <p>
        Please enter your email address and password below.
      </p>
    </div>
  </div>
  <div
    className="sm-spacer validation-error"
  >
    <h5>
      Please enter a valid email
    </h5>
  </div>
  <form
    onSubmit={[Function]}
  >
    <div
      className="row md-spacer"
    >
      <div
        className="col-sm-6 input-row"
      >
        <div />
      </div>
      <div
        className="col-sm-6 input-row"
      >
        <input
          className="form-group hg-input v2"
          id="loginPassword"
          label="Your Password"
          placeholder="Your Password"
          required={true}
          type="password"
        />
      </div>
    </div>
    <div
      className="row lg-spacer"
    >
      <div
        className="col-xs-offset-3 col-xs-6 flex-column"
      >
        <div />
        <div
          className="row sm-spacer"
        >
          <div
            className="remember-checkbox"
          >
            <div
              className="checkaroo"
            >
              <i
                aria-hidden="true"
                className="fa fa-check"
              />
            </div>
          </div>
          <label>
            Remember Me
          </label>
        </div>
        <div
          className="row md-spacer forgot-password"
        >
          <a
            href="/resetpassword"
            onClick={[Function]}
          >
            Forgot your password?
          </a>
        </div>
      </div>
    </div>
  </form>
</div>
`;

exports[`Sign On Page Elements 1`] = `
<div
  className="sign-on-wrapper"
>
  <div
    className="row"
  >
    <h2>
      SIGN ON
    </h2>
    <div
      className="page-instructions col-xs-offset-4 col-xs-4"
    >
      <p>
        Please enter your email address and password below.
      </p>
    </div>
  </div>
  <form
    onSubmit={[Function]}
  >
    <div
      className="row md-spacer"
    >
      <div
        className="col-sm-6 input-row"
      >
        <div />
      </div>
      <div
        className="col-sm-6 input-row"
      >
        <input
          className="form-group hg-input v2"
          id="loginPassword"
          label="Your Password"
          placeholder="Your Password"
          required={true}
          type="password"
        />
      </div>
    </div>
    <div
      className="row lg-spacer"
    >
      <div
        className="col-xs-offset-3 col-xs-6 flex-column"
      >
        <div />
        <div
          className="row sm-spacer"
        >
          <div
            className="remember-checkbox"
          >
            <div
              className="checkaroo"
            >
              <i
                aria-hidden="true"
                className="fa fa-check"
              />
            </div>
          </div>
          <label>
            Remember Me
          </label>
        </div>
        <div
          className="row md-spacer forgot-password"
        >
          <a
            href="/resetpassword"
            onClick={[Function]}
          >
            Forgot your password?
          </a>
        </div>
      </div>
    </div>
  </form>
</div>
`;
